### ISSUE TITLE
Bug with storage_profile enabled at creation.

### ISSUE THREAD

**Issue by nakasix** [vmware/terraform-provider-vcd#754]
Hello,

When I set the "enabled" value to "false" of the "storage_profile" block of resource "vcd_org_vdc", the value is not taken into account when creating the VDC.

The output of Terraform:
![image](https://user-images.githubusercontent.com/37986982/144067626-7bdd8a16-60fd-4b7c-bf37-6627523f5306.png)

The result in VCD GUI:
![image](https://user-images.githubusercontent.com/37986982/144067872-7ba7760a-825e-4eaa-bf8c-701dee0ce6a6.png)

NB: However, once the VDC is created and the module is reread, the variable is taken in consideration.

Regards,

**Comment by vbauzysvmware** [comment]
Hello, @nakasix 

at the start could provide VCD and terraform provider version ?

**Comment by nakasix** [comment]
Hello @vbauzysvmware ,

Of course!

VMware Cloud Director:
`10.3.1.18738571`

Terraform:
`{
  "terraform_version": "1.0.11",
  "platform": "linux_amd64",
  "provider_selections": {
    "registry.terraform.io/vmware/vcd": "3.4.0"
  },
  "terraform_outdated": false
}`

Regards,

**Comment by vbauzysvmware** [comment]
Also full HCL :) if need you can obfuscate details.

**Comment by nakasix** [comment]
Hello,

What do you mean by HCL?

- The .terraform.lock.hcl file?
- The terraform main.tf module?
- Hardware Compatibility List? :)

Regards,

**Comment by nakasix** [comment]
Hello,

I did not put the declaration of my variables for more readability.
```
terraform {
  required_providers {
    vcd = {
      source = "vmware/vcd"
    }
  }
}

provider "vcd" {
  user     = var.vcd_user
  password = var.vcd_password
  org      = "System"
  url      = var.vcd_url
}

resource "vcd_org_vdc" "vcd_org_vdc" {

  name = var.vcd_org_vdc
  org  = var.vcd_org

  allocation_model           = "Flex"
  elasticity                 = true
  include_vm_memory_overhead = false

  network_pool_name = var.network_pool_name
  provider_vdc_name = var.provider_vdc_name

  cpu_speed = 2000

  cpu_guaranteed    = 0
  memory_guaranteed = 0

  network_quota = 1000

  compute_capacity {
    cpu {
      allocated = tonumber(var.compute_capacity_cpu) * 2000
      limit     = 0
    }

    memory {
      allocated = tonumber(var.compute_capacity_memory) * 1024
      limit     = 0
    }
  }

  storage_profile {
    name    = "SSD"
    enabled = tonumber(var.storage_profile_ssd) != 0 ? true : false
    limit   = tonumber(var.storage_profile_ssd) * 1024
    default = tonumber(var.storage_profile_ssd) != 0 ? true : false
  }

  storage_profile {
    name    = "SATA"
    enabled = tonumber(var.storage_profile_sata) != 0 ? true : false
    limit   = tonumber(var.storage_profile_sata) * 1024
    default = tonumber(var.storage_profile_sata) != 0 && tonumber(var.storage_profile_ssd) == 0 ? true : false
  }

  enabled                  = true
  enable_thin_provisioning = true
  enable_fast_provisioning = false
  delete_force             = true
  delete_recursive         = true
}
```

Regards,

**Comment by vbauzysvmware** [comment]
Solved and fix will be released in 3.6.
